tools/authors.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python2
+# -*- coding: utf-8 -*-
+# Copyright (c) 2017, Niklas Hauser
+# All rights reserved.
+#
+# The file is part of the xpcc library and is released under the 3-clause BSD
+# license. See the file `LICENSE` for the full license governing this code.
+# -----------------------------------------------------------------------------
+
+import subprocess
+
+# store the commits by author
+shortlog = {}
+
+# get the shortlog summary
+output = subprocess.Popen("git shortlog -sn", shell=True, stdout=subprocess.PIPE).stdout.read()
+# parse the shortlog
+for line in output.split("\n")[:-1]:
+    commits, author = line.split("\t")
+    shortlog[author] = int(commits)
+
+# some authors have aliases cos estupido
+author_alias = {
+    "Sascha Schade": [
+        "strongly-typed",
+        "Webb"
+    ],
+    "Kevin Läufer": [
+        "Laeufer",
+        "Kiwi",
+        "Kevin"
+    ],
+    "Daniel Krebs": [
+        "daniel"
+    ],
+    "Niclas Rohrer": [
+        "Niclas"
+    ],
+    "Martin Rosekeit": [
+        "martin",
+        "thundernail"
+    ],
+    "Georgi Grinshpun": [
+        "Georgi"
+    ],
+    "Thorsten Lajewski": [
+        "thorsten"
+    ],
+    "Hans Schily": [
+        "hans"
+    ]
+}
+
+# cleaned up dictionary
+commit_count = {}
+
+def add_commits(author, commits):
+    if author not in commit_count:
+        commit_count[author] = 0
+    commit_count[author] += commits
+
+# merge authors
+for slauthor in shortlog.keys():
+    for author, aliases in author_alias.items():
+        if any(name in slauthor for name in (aliases + [author])):
```

Comment:
That doesn't seem to be too robust. What if someone else named Kevin (or any of the aliases) contributes? That would match the alias incorrectly.
As for `[author]`: you wouldn't need to add the "canonical" name to the aliases, since if it is not found in the alias list, the for loop would fall over to the else branch, and add the author there (as for every non-aliased author).